chapter4/astroml_chapter4_Nonparametric_modeling_and_selection_effects.ipynb

```diff
@@ -0,0 +1,740 @@
+    "                        linestyle='dashed', lw=1, alpha=0.5))\n",
+    "ax2.text(0.5 * max_func(yi), 0.5 * yi, '$J_k$', ha='center', va='center', fontsize = 14)\n",
+    "\n",
+    "# adjust the limits of both axes\n",
+    "for ax in (ax1, ax2):\n",
+    "    ax.set_xlim(0, 1.1)\n",
+    "    ax.set_ylim(0, 1.1)\n",
+    "    ax.set_xlabel('$x$', fontsize = 14)\n",
+    "    ax.set_ylabel('$y$', fontsize = 14)\n",
+    "\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "68ea103c",
+   "metadata": {},
+   "source": [
+    "In general, this problem can be solved by fitting some predefined (assumed) function to the data (i.e., determining a set of best-fit parameters) or in a nonparametric way. An elegant nonparametric solution to this mathematical problem was developed by Lynden-Bell, and shown to be equivalent to or better than other nonparametric methods. Lynden-Bell's method belongs to methods known in statistical literature as product-limit estimators.\n",
+    "\n",
+    "### Lynden-Bell's $C^-$ method\n",
+    "\n",
+    "Lynden-Bell's nonparametric $C^−$ method can be applied to the above problem when the distributions along the two coordinates $x$ and $y$ are uncorrelated, that is, when we can assume that the bivariate distribution $n(x, y)$ is separable:\n",
+    "\n",
+    "$$ n(x,y) = \\Psi(x)\\rho(y) $$\n",
+    "\n",
+    "Therefore, before using the $C^−$ method, we need to demonstrate that this assumption is valid.\n",
+    "\n",
+    "Following Lynden-Bell, the basic steps for testing that the bivariate distribution $n(x, y)$ is separable are the following:\n",
+    "\n",
+    "\n",
+    ">1. Define a comparable or associated set for each object $i$ such that $J_i = \\{j: x_j < x_i, y_j < y_\\text{max}(x_i)\\}$; this is the largest $x$-limited and $y$-limited data subset for object $i$, with $N_i$ elements (see the left panel of the figure above).\n",
+    ">\n",
+    ">2. Sort the set $J_i$ by $y_j$; this gives us the rank $R_j$ for each object (ranging from 1 to $N_i$).\n",
+    ">\n",
+    ">3. Define the rank $R_i$ for object $i$ in its associated set: this is essentially the number of objects\n",
+    "with $y<y_i$ in set $J_i$.\n",
+    ">\n",
+    ">4. Now, if $x$ and $y$ are truly independent, $R_i$ must be distributed uniformly between 0 and $N_i$; in this case, it is trivial to determine the expectation value and variance for $R_i: E(R_i) = E_i = N_i/2$ and $V(R_i) = V_i = N_i^2/12$. We can define the statistic $$ \\tau = \\frac{\\sum_i(R_i-E_i)}{\\sqrt{\\sum_iV_i}} $$ If $\\tau$ < 1, then $x$ and $y$ are uncorrelated at $\\sim 1\\sigma$ level (this step appears similar to Schmidt’s $V/V_\\text{max}$ test discussed below; nevertheless, they are fundamentally different because $V/V_\\text{max}$ tests the hypothesis of a uniform distribution in the $y$ direction, while the statistic $\\tau$ tests the hypothesis of uncorrelated $x$ and $y$).\n",
```

Comment:
there is a rendering issue in this line, but the whole list looks a bit off. This should be fixed before merging.

<img width="827" alt="Screenshot 2023-05-30 at 16 41 24" src="https://github.com/astroML/astroML-notebooks/assets/6788290/7d8d2b11-eede-4f7f-9887-2d1a8921efd7">